src/main/java/seedu/address/model/deck/Card.java

```diff
@@ -20,32 +31,75 @@ public Card(Question question, Answer answer) {
         requireAllNonNull(question, answer);
         this.question = question;
         this.answer = answer;
-        performance = Performance.NORMAL;
+        this.reviewScore = DEFAULT_REVIEW_SCORE;
         timesReviewed = 0;
+        performance = Performance.NORMAL;
+        this.nextReviewDate = LocalDateTime.now();
     }
 
-    public Card(Question question, Answer answer, Performance performance, int timesReviewed) {
-        requireAllNonNull(question, answer, performance, timesReviewed);
+    public Card(Question question, Answer answer, Performance performance, int timesReviewed, double reviewScore,
+                LocalDateTime nextReviewDate) {
+        requireAllNonNull(question, answer);
         this.question = question;
         this.answer = answer;
         this.performance = performance;
         this.timesReviewed = timesReviewed;
+        this.nextReviewDate = nextReviewDate;
+        this.reviewScore = reviewScore;
     }
-
     public Card(Card other) {
         this.question = new Question(other.getQuestion().toString());
         this.answer = new Answer(other.getAnswer().toString());
         this.performance = other.performance;
         this.timesReviewed = other.timesReviewed;
+        this.nextReviewDate = other.nextReviewDate;
+        this.reviewScore = other.reviewScore;
     }
 
+    /**
+     * Assigns performance to the currently reviewed card.
+     * @param card the card being reviewed
+     * @param performance how well the user remembers the card
+     */
     public static Card classifyCard(Card card, Performance performance) {
-        return new Card(card.question, card.answer, performance, card.timesReviewed + 1);
+        int performanceAsInt = performance.ordinal();
+        double scorePostUpdate = card.updateReviewScore(performanceAsInt);
+        LocalDateTime nextReviewDate = calculateNextReviewDate(card, performance);
+        return new Card(card.question, card.answer, performance, card.timesReviewed + 1,
+                scorePostUpdate, nextReviewDate);
+    }
+    /**
+     *  Find out when the card needs to be reviewed again for optimal recall.
+     */
+    private static LocalDateTime calculateNextReviewDate(Card card, Performance performance) {
+        double consecutiveCorrectAnswers = card.getConsecutiveCorrect();
+        double addedDays = 1;
+        if (performance.isCorrect()) {
+            addedDays = REVIEW_INTERVAL_COEFFICIENT * Math.pow(card.reviewScore, consecutiveCorrectAnswers - 1);
+        }
+        System.out.println("addedDays " + addedDays);
```

Comment:
Is it for debugging? Should remove after done with debugging.

Reply:
Yup, I missed that. Thanks for pointing this out!